src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -392,24 +428,20 @@ internal CompilationWithAnalyzers GetCompilationWithAnalyzers(Project project, C
             }
         }
 
-        private bool SkipRunningAnalyzer(
-            CompilationOptions compilationOptions,
-            DiagnosticAnalyzerDriver userDiagnosticDriver,
-            bool openedDocument,
-            bool skipClosedFileChecks,
-            StateSet stateSet)
+        private async Task<bool> SkipRunningAnalyzerAsync(Project project, DiagnosticAnalyzer analyzer, bool openedDocument, bool skipClosedFileCheck, CancellationToken cancellationToken)
         {
-            if (Owner.IsAnalyzerSuppressed(stateSet.Analyzer, userDiagnosticDriver.Project))
+            // this project is not in a good state. only continue if it is opened document.
+            if (!await project.HasSuccessfullyLoadedAsync(cancellationToken).ConfigureAwait(false) && !openedDocument)
```

Comment:
What happens when project had loaded successfully, had diagnostics and then goes into this bad state. Shouldn't we clear the diagnostics and then skip running analyzers?